### PR TITLE
Add RedCaps dataset

### LINKED ISSUE #3316: Add RedCaps dataset
## Adding a Dataset
- **Name:** RedCaps
- **Description:** Web-curated image-text data created by the people, for the people
- **Paper:** https://arxiv.org/abs/2111.11431
- **Data:** https://redcaps.xyz/
- **Motivation:** Multimodal image-text dataset:  12M+ Image-text pairs 

Instructions to add a new dataset can be found [here](https://github.com/huggingface/datasets/blob/master/ADD_NEW_DATASET.md).

Proposed by @patil-suraj

### PR DESCRIPTION
**Pull request by mariosasko**
Add the RedCaps dataset. I'm not adding the generated `dataset_infos.json` file for now due to its size (11 MB). 

TODOs:
- [x] dummy data
- [x] dataset card

Close #3316

### CODE REVIEW

**Comment by lhoestq** [comment]
Cool ! If you want you can include `dataset_infos.json` but only for the main configurations. That's what we do for example for translation datasets when there are too many configs

**Review by lhoestq — APPROVED** [review]
Awesome !

Maybe add a code example that shows how to to download the images using `map` ?

**Comment by mariosasko** [comment]
@lhoestq I've added an example that uses `map` to download the images.

File: datasets/red_caps/README.md [hunk 1]
```diff
@@ -68,6 +69,41 @@ fine-grained descriptions (northern cardinal, taj mahal). Subreddit names provid
...
+    import requests
+
+    images = []
+    for image_url in batch["image_url"]:
+        try:
+            image = PIL.Image.open(
+                requests.get(
+                    image_url,
+                    stream=True,
+                    headers={"user-agent": get_datasets_user_agent()},
```
Comment (lhoestq) [inline datasets/red_caps/README.md:91]: not sure the user agent is necessary but ok :)

**Review by lhoestq — APPROVED** [review]
Thanks ! :)

File: datasets/red_caps/README.md (hunk 1 above)
Comment (mariosasko) [reply datasets/red_caps/README.md:91]: This fixes the "429: Too many requests" HTTP error that I'd get when trying to download the images on Colab (I can't reproduce this issue locally). Also, img2dataset, a popular tool for building image datasets from image urls, specifies `user-agent` in its [downloader](https://github.com/rom1504/img2dataset/blob/83a757976f4e0ad719068029b2fd41d5a409d13a/img2dataset/downloader.py#L27).

**Pull request merged**